Signal/src/view controllers/CallViewController.swift

```diff
@@ -506,6 +528,37 @@ class CallViewController: UIViewController, CallDelegate {
 
         default: break
         }
+
+        if callState == .connected {
+            if callDurationTimer == nil {
+                let kDurationUpdateFrequencySeconds = 1 / 20.0
+                callDurationTimer = Timer.scheduledTimer(timeInterval: kDurationUpdateFrequencySeconds,
+                                                         target:self,
+                                                         selector:#selector(updateCallDuration),
+                                                         userInfo:nil,
+                                                         repeats:true)
+            }
+        } else {
+            callDurationTimer?.invalidate()
+            callDurationTimer = nil
+        }
+    }
+
+    func updateCallDuration(timer: Timer?) {
+        updateCallUI(callState: call.state)
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
```

Comment:
style nit: I like to keep the view lifecyle methods together at the top, after any inits.